pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -526,6 +528,40 @@ func (d *namespacedResourcesDeleter) deleteAllContent(ctx context.Context, ns *v
 		gvrToNumRemaining:        map[schema.GroupVersionResource]int{},
 		finalizersToNumRemaining: map[string]int{},
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.OrderedNamespaceDeletion) {
+		// TODO: remove this log when the feature gate is enabled by default
+		logger.V(5).Info("Namespace controller - OrderedNamespaceDeletion feature gate is enabled", "namespace", namespace)
+		// Ensure all pods in the namespace are deleted first
+		podsGVR := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+		gvrDeletionMetadata, err := d.deleteAllContentForGroupVersionResource(ctx, podsGVR, namespace, namespaceDeletedAt)
+		if err != nil {
+			if err != nil {
```

Comment:
nit: Error handling is being done twice here.

Reply:
Updated. Thanks for catching this!